### DISCUSSION TITLE
Issue with Gitlab CI setup with Reference architecture

### DISCUSSION THREAD

**Discussion by velarx** [gruntwork-io/knowledge-base#597] [How do I]
Currently gruntwork team had setup the reference architecture and mentioned the steps to integrate the CI pipeline. Getting the error within the CI for while invoking the lambda function. One thing I has changed in the common.hcl and deploy-infra.sh for the Infrastructure-live https and ssh URL.
![Screenshot_1](https://user-images.githubusercontent.com/91253270/203539777-169b5e71-9aa9-4e71-947e-dc2f070eaf0d.png)

---

<ins datetime="2022-11-23T11:50:46Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109640">Tracked in ticket #109640</a></p>
</ins>

**Comment by bethadele** [comment]
I'm forwarding this to one of our Reference Architecture experts to take a look.

**Comment by denis256** [comment]
Hi,
looking at the error message - looks like is allowed only git repositories accessible over SSH via `git@github...` but during invocation was passed git repository url which starts with `http://`

So or update allowed pattern to work with `http://` or pass SSH address of repository(and have configured SSH keys to pull repository)

**Reply by ivishalvarshney** [reply]
Hello @denis256 ,

Thanks for you reply. But let me explain you our scenario in more details,
We have enable CI/CD pipeline through gitlab. While cloning the code, we also created another branch (for eg:- branchA) to make the changes. Our objective is, once our changes will run in the pipline at plan stage we will merge the change into master branch and pipeline will run terragrunt apply.
We have changed the url from github to gitlab (in deploy-infra.sh and common.hcl), even after changing/hardcoding url (in ecs-deploy-runner.hcl) our pipeline in another branch (branchA), it is throwing the "OptionRegexRestrictedError". The regex is formed using the original URL of github, and our new url is of gitlab.

Please help to resolve this error.

**Reply by denis256** [reply]
Hi,
I'm not sure what was changed in `common.hcl / deploy-infra.sh`
I would suggest applying incremental changes:
* apply separated changes in the whitelisted git repo with `additional_plan_and_apply_repos` having gitlab repos
* synchronize changes to gitlab repo and deploy from there
